client/client.go

```diff
@@ -117,6 +128,6 @@ func handleErrResp(resp *http.Response) string {
 }
 
 // Helper method to generate error message for client functions
-func genRespErr(ep string, err string) error {
-	return fmt.Errorf("error calling %s endpoint - %s", ep, err)
+func genRespErr(code int, ep, err string) error {
```

Comment:
small nit here - can we use something like this? (pseudo-code):
```go
func genRespErr(resp *http.Response, ep, err string) error {
  if err == "" { // or maybe we can use err == nil?
    err := handleErrResp(resp)
  }

  return ApiErr{resp.StatusCode, ep, error}
}
```

Reply:
@unleashed I've pushed some changes here